File: Python-SDK/MySplit.py
from splitio import get_factory
from splitio.exceptions import TimeoutException
import logging

class MySplit:
    def __init__(self, APIKey):
        try:
            factory = get_factory(APIKey, config={'impressionsRefreshRate' : 5, 'eventsPushRate': 5})
            factory.block_until_ready(5)
            self.splitClient = factory.client()
        except Exception:
            print("an exception")

    def getSplitTreatment(self, clientId, splitName, attributes):
        if attributes==None:
            return self.splitClient.get_treatment(clientId, splitName)
        else:
            return self.splitClient.get_treatment(clientId, splitName, attributes)

    def getSplitTreatmentWithConfig(self, clientId, splitName, attributes):
        if attributes==None:
            return self.splitClient.get_treatment_with_config(clientId, splitName)
        else:
            return self.splitClient.get_treatment_with_config(clientId, splitName, attributes)

    def sendTrack(self, clientId, trackType, eventType):
        return self.splitClient.track(clientId, trackType, eventType)
    
    def __del__(self):
        self.splitClient.destroy()

    def destroy(self):
        self.splitClient.destroy()

File: Java-SDK/SplitSDK_Java/src/sample/MySplit.java
package sample;

import java.util.List;
import java.util.Map;

import io.split.client.SplitClient;
import io.split.client.SplitClientConfig;
import io.split.client.SplitFactory;
import io.split.client.SplitFactoryBuilder;
import io.split.client.api.SplitResult;

public class MySplit {
	SplitClient client;
	SplitFactory splitFactory;

	MySplit(String APIKey) throws Exception {
		SplitClientConfig config = SplitClientConfig.builder()
				.setBlockUntilReadyTimeout(10000)
				.numThreadsForSegmentFetch(6)
//				.enableDebug()
				.build();
		try {
		    this.splitFactory = SplitFactoryBuilder.build(APIKey, config);
		    
		    this.client = this.splitFactory.client();
		    this.client.blockUntilReady();
			System.out.println("*************After blockUntilReady*********");
		    
		    
		} catch (Exception e) {
			System.out.print("Exception: "+e.getMessage());
		}
	}

	SplitResult GetSplitTreatmentWithConfig(String clientId, String splitName, Map<String, Object> attributes) {
        return this.client.getTreatmentWithConfig(clientId, splitName, attributes);				
	}
	SplitResult GetSplitTreatmentWithConfig(String clientId, String splitName) {
        return this.client.getTreatmentWithConfig(clientId, splitName);				
	}
	String GetSplitTreatment(String clientId, String splitName, Map<String, Object> attributes) {
        return this.client.getTreatment(clientId, splitName, attributes);				
	}
	String GetSplitTreatment(String clientId, String splitName) {
        return this.client.getTreatment(clientId, splitName);				
	}
	boolean SendTrackEvent(String userId, String trackType, String metricName, double metricValue) {
		return this.client.track(userId, trackType, metricName, metricValue);
	}
	boolean SendTrackEvent(String userId, String trackType, String metricName) {
		return this.client.track(userId, trackType, metricName);
	}	
	boolean SendTrackEvent(String userId, String trackType, String metricName, Map<String, Object> properties) {
		return this.client.track(userId, trackType, metricName, properties);
	}	
	boolean SendTrackEvent(String userId, String trackType, String metricName, double metricValue, Map<String, Object> properties) {
		return this.client.track(userId, trackType, metricName, metricValue, properties);
	}	
	List<String> GetSplitNames() {
        return this.splitFactory.manager().splitNames();				
	}

	void Destroy()  {
		this.client.destroy();
	}
	
	protected void finalize ()  {
		this.client.destroy();
	}

}
